packages/core/src/browser/quick-input/quick-command-frontend-contribution.ts

```diff
@@ -13,33 +13,29 @@
  *
  * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
  ********************************************************************************/
-
 import { injectable, inject } from 'inversify';
-import { Command, CommandRegistry, CommandContribution, MenuContribution, MenuModelRegistry } from '../../common';
+import { CommandRegistry, CommandContribution, MenuContribution, MenuModelRegistry } from '../../common';
 import { KeybindingRegistry, KeybindingContribution } from '../keybinding';
-import { PrefixQuickOpenService, QuickOpenHandlerRegistry } from './prefix-quick-open-service';
 import { CommonMenus } from '../common-frontend-contribution';
-
-export const quickCommand: Command = {
-    id: 'workbench.action.showCommands'
-};
-
-export const CLEAR_COMMAND_HISTORY: Command = {
-    id: 'clear.command.history',
-    label: 'Clear Command History'
-};
+import { QuickInputService } from './quick-input-service';
+import { CLEAR_COMMAND_HISTORY, quickCommand, QuickCommandService } from './quick-command-service';
+import { QuickInputContribution } from './quick-input-contribution';
 
 @injectable()
-export class QuickCommandFrontendContribution implements CommandContribution, KeybindingContribution, MenuContribution {
+export class QuickCommandFrontendContribution implements CommandContribution, KeybindingContribution, MenuContribution, QuickInputContribution {
 
-    @inject(PrefixQuickOpenService)
-    protected readonly quickOpenService: PrefixQuickOpenService;
+    @inject(QuickInputService)
+    protected readonly quickInputService: QuickInputService;
 
-    @inject(QuickOpenHandlerRegistry) protected readonly quickOpenHandlerRegistry: QuickOpenHandlerRegistry;
+    @inject(QuickCommandService)
+    protected readonly quickCommandService: QuickCommandService;
 
     registerCommands(commands: CommandRegistry): void {
         commands.registerCommand(quickCommand, {
-            execute: () => this.quickOpenService.open('>')
+            execute: () => {
+                this.quickCommandService.init();
```

Comment:
Also, why does quickCommandService need this call and other contributors to quick access don't?

Reply:
> Also, why does quickCommandService need this call and other contributors to quick access don't?

In Theia 0.20.x, there were different implementations and usages for quickOpen. The "OnType" type functionality would invoke `init` method if that existed (see screen cap below). So in the 0.23.x commands quickInput implementation I added an 'init" function in same manner. One thing to note is that command 'workbench.action.showCommands' is triggered by executing that command and not from shortcut or from Ctrl+ P >> ? >> selecting '>'. So it's not running this code all the time. 

![image](https://user-images.githubusercontent.com/71069170/117812508-8a35ef00-b26a-11eb-979d-08c3693778b4.png)

Reply:
> `init` sounds a lot like a one-time setup, but this is called every time we open the `F1` menu. Maybe we should rename to "reset" or something similar?

![image](https://user-images.githubusercontent.com/71069170/117812508-8a35ef00-b26a-11eb-979d-08c3693778b4.png)
As you can see here in 0.20.x it was run for each quickOpen was opened but will change it to `reset` as requested.

Reply:
Fixed

Reply:
So as I understand it, the "reset()" method should give quick access providers a chance to initialize before the quick access UI is shown. I think that functionality makes sense: we should extend it to all quick access providers, not just quick command service.

Reply:
> So as I understand it, the "reset()" method should give quick access providers a chance to initialize before the quick access UI is shown. I think that functionality makes sense: we should extend it to all quick access providers, not just quick command service.

I moved 'reset' back to `quick-input-service.ts` - so that all services derived from it could theoretically add a reset function. Is that what you thought of?

Reply:
Yes, ever contributor should have the option to do pre-open initialization.